src/components/LoginFrom.jsx

```diff
@@ -0,0 +1,125 @@
+import "./Required.css";
+import { Link, useNavigate } from "react-router-dom";
+import { useState } from "react";
+
+function LoginForm() {
+  const [username, setUsername] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [errors, setErrors] = useState({
+    username: "",
+    email: "",
+    password: "",
+  });
+
+  const validateForm = () => {
+    const newErrors = { username: "", email: "", password: "" };
+    let hasErrors = false;
+
+    if (!username) {
+      newErrors.username = "ユーザー名を入力してください。";
+      hasErrors = true;
+    }
+
+    if (!email) {
+      newErrors.email = "メールアドレスを入力してください。";
+      hasErrors = true;
+    }
+
+    if (!password) {
+      newErrors.password = "パスワードを入力してください。";
+      hasErrors = true;
+    }
+
+    setErrors(newErrors);
+    return !hasErrors;
```

Comment:
わざわざ反転させて返す理由が薄いね。これなら普通にbooleanとして返したほうが読みやすいし、このfunctionを使うときに予想しやすいのでこっちのがお勧め。
```suggestion
    return hasErrors;
```